src/context.c

```diff
@@ -315,6 +316,15 @@ static gboolean r_context_configure_target(GError **error)
 		}
 	}
 
+	/* load global state */
+	g_clear_pointer(&context->statepath, g_free);
+	g_clear_pointer(&context->global_state, r_global_state_free);
+	context->statepath = g_build_filename(context->config->data_directory, "state", NULL);
```

Comment:
`data_directory` can be unset. Perhaps move it inside the `if` above?

Also, `state` as a filename seems too generic. Perhaps something mentioning persistence?

Reply:
We discussed that due to atomicity using the existing status file should be preferred.
Then we still need to check for per-slot not being enabled.

I'll implement it this way but rethinking your proposal I am not sure if atomicity plays any role since the update of the slot status and the update of the system status would never happen at the same time I guess.